internals/updates/docker.go

```diff
@@ -0,0 +1,59 @@
+package updates
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/portainer/agent"
+)
+
+type DockerUpdaterCleaner struct {
+}
+
+func NewDockerUpdaterCleaner() *DockerUpdaterCleaner {
+	return &DockerUpdaterCleaner{}
+}
+
+func (du *DockerUpdaterCleaner) Clean(ctx context.Context) error {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion(agent.SupportedDockerAPIVersion))
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	foundRunningContainer := false
+	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{All: true})
+	if err != nil {
+		return fmt.Errorf("failed to list containers: %s", err.Error())
+	}
+	for _, container := range containers {
+		_, ok := container.Labels["io.portainer.hideStack"]
```

Comment:
didn't we decide to use a different label? something like `io.portainer.updater=true`?  `hideStack` can be used to hide other stacks, so it might return wrong containers. 

also, why not call `cli.ContainerList` with the label filter?

Reply:
Although we had previously made the decision to update the server from CE to BE, the agent update was merged before that decision was made. As a result, I will update here accordingly to ensure consistency in the implementation.